features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddOrEditDesignFragment.kt

```diff
@@ -0,0 +1,294 @@
+
+  private fun openGallery() {
+    val galleryIntent = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI).apply {
+      type = Constants.TYPE_IMAGE_ALL
+    }
+    this.startActivityForResult(galleryIntent, GALLERY_REQUEST_CODE)
+  }
+
+  private fun openAddColorBottomSheet(name: String? = null, color: String? = null) {
+    AddColorBottomSheetFragment.newInstance(::addColorChip, name, color).show(parentFragmentManager,
+        getScreenName())
+  }
+
+  private fun openAddSizeBottomSheet(name: String? = null, sizeDetail: SizeDetailUiModel? = null) {
+    AddSizeBottomSheetFragment.newInstance(::addSizeChip, name, sizeDetail).show(
+        parentFragmentManager, getScreenName())
+  }
+
+  private fun setClickable(value: Boolean) {
+    with(binding.layoutAddOrEditImage.root) {
+      isClickable = value
+      isFocusable = value
+      isFocusableInTouchMode = value
+    }
+  }
+
+  private fun setData(response: DesignDetailResponse) {
+    with(binding) {
+      editTextDesignName.setText(response.title)
+      editTextDesignPrice.setText(response.price.toString())
+      editTextDesignDiscount.setText(response.discount.toString())
+      editTextDesignDescription.setText(response.description)
+    }
+    showImagePreview(response.image, response.title)
+    setSize(response.size)
+    setColor(response.color)
+  }
+
+  private fun setColor(colors: List<ColorResponse>) {
+    colors.forEach {
+      addColorChip(it.id, it.color)
+    }
+  }
+
+  private fun setSize(sizes: List<SizeResponse>) {
+    sizes.forEach { size ->
+      size.detail?.let {
+        addSizeChip(size.id, DesignDetailMapper.mapToSizeDetailUiModel(it))
+      }
+    }
+  }
+
+  private fun showImagePreview(imageUri: Uri, imageName: String) {
+    showImagePreview(imageName)
+    setImage(imageUri)
+  }
+
+  private fun showImagePreview(imageUrl: String, imageName: String) {
+    showImagePreview(imageName)
+    setImage(imageUrl)
+  }
+
+  private fun showImagePreview(imageName: String) {
+    with(binding.layoutAddOrEditImage) {
+      groupFilledImageState.show()
+      groupEmptyImageState.remove()
+      textViewPreviewDesignImageName.text = imageName
+    }
+    setClickable(false)
+  }
+
+  private fun setImage(imageUrl: String) {
+    context?.let { context ->
+      ImageLoader.loadImageUrl(context, imageUrl,
+          binding.layoutAddOrEditImage.imageViewPreviewDesignImage)
+    }
+  }
+
+  private fun setImage(imageUri: Uri) {
+    binding.layoutAddOrEditImage.imageViewPreviewDesignImage.setImageURI(imageUri)
+  }
+
+  private fun validate() {
+    with(binding) {
+      val name = editTextDesignName.text()
+      val price = editTextDesignPrice.text()
+      val discount = editTextDesignDiscount.text()
+      val description = editTextDesignDescription.text()
+
+      if (name.isNotBlank() && price.isNotBlank() && discount.isNotBlank() && description.isNotBlank() && viewModel.isPriceValid(
```

Comment:
put this 
name.isNotBlank() && price.isNotBlank() && discount.isNotBlank() && description.isNotBlank() && viewModel.isPriceValid(price, discount)
in new method e.g. 
fun isPriveValid(price: Double, discount: Double) = name.isNotBlank() && price.isNotBlank() && discount.isNotBlank() && description.isNotBlank() && ...